tests/test_subsector.py

```diff
@@ -105,20 +104,7 @@ def test_subsector_noninvesting_aggregation(market, model, technologies, tmp_pat
         commodity=technologies.commodity, region=technologies.region
     ).interp(year=[2020, 2025])
     assert all(agent.year == 2020 for agent in agents)
-    result = subsector.aggregate_lp(technologies, market)
```

Comment:
Why was this returning something different that `None`? I think this test only makes sense if `aggregate_lp` returns something different than `None`, which we have seen cannot happen. And if it can happen, as this test was demonstrating, under what conditions? Are we removing functionality that is actually useful (not necessarily in use)?

Reply:
When using the `Agent` class, `aggregate_lp` would return the search space and constraints, and then these would be used to perform investments withing `Subsector.invest`. Conversely, the `InvestingAgent` performs investments within its own `next` method. In theory the net result should be the same, so I can't see any reason to use `Agent` over `InvestingAgent`. I'm guessing it's just a legacy thing that was never tidied up